src/Concerns/RegistersEventListeners.php

```diff
@@ -17,34 +20,27 @@ trait RegistersEventListeners
      */
     public function registerEvents(): array
     {
+        $listenersClasses = [
+            BeforeExport::class,
+            BeforeWriting::class,
+            BeforeImport::class,
+            AfterImport::class,
+            AfterBatch::class,
+            AfterChunk::class,
+            AfterChunk::class,
+            ImportFailed::class,
+            BeforeSheet::class,
+            AfterSheet::class,
+        ];
         $listeners = [];
 
-        if (method_exists($this, 'beforeExport')) {
-            $listeners[BeforeExport::class] = [static::class, 'beforeExport'];
-        }
-
-        if (method_exists($this, 'beforeWriting')) {
-            $listeners[BeforeWriting::class] = [static::class, 'beforeWriting'];
-        }
-
-        if (method_exists($this, 'beforeImport')) {
-            $listeners[BeforeImport::class] = [static::class, 'beforeImport'];
-        }
-
-        if (method_exists($this, 'afterImport')) {
-            $listeners[AfterImport::class] = [static::class, 'afterImport'];
-        }
-
-        if (method_exists($this, 'importFailed')) {
-            $listeners[ImportFailed::class] = [static::class, 'importFailed'];
-        }
-
-        if (method_exists($this, 'beforeSheet')) {
-            $listeners[BeforeSheet::class] = [static::class, 'beforeSheet'];
-        }
-
-        if (method_exists($this, 'afterSheet')) {
-            $listeners[AfterSheet::class] = [static::class, 'afterSheet'];
+        foreach ($listenersClasses as $class) {
```

Comment:
I prefer to keep the code as is

Reply:
Maybe like this, without the reflection, it at least prevents the repetition